### PR TITLE
Change how assembly quick pick is displayed (ilspy.decompileAssemblyInWorkspace)

### LINKED ISSUE #61: Difficult to pick assembly to decompile in current workspace: filenames are too long
As you can see on the screenshot below, the only thing I can see is the base path of the file, which is troublesome. It should be possible to show a relative path or at least make a configuration option for it.

image

I would be happy to contribute, please let me know if you accept contributions.

### PR DESCRIPTION
**Pull request by NikiforovAll**
Closes #61

### CODE REVIEW

**Comment by jeremymeng** [comment]
Awesome!  Do you have a screenshot after this change?  I will review this PR late today.

**Comment by NikiforovAll** [comment]
Sure thing! 
il-spy-ext-demo

File: vscode-extension/src/extension.ts [hunk 1]
```diff
@@ -173,3 +176,51 @@ async function promptForAssemblyFilePathViaDialog(): Promise<string> {
...
+          relativePath: p.dir.replace(assemblyWorkspace.uri.fsPath, ''),
+          workspaceFolder: assemblyWorkspace.name
+        };
+    });
+}
+
+function createAssemblyQuickPickItem(assemblyPathInfo: AssemblyPathInfo): vscode.QuickPickItem{
+    const selectIcon = (extension: string) => {;
+      switch (extension) {
+        case '.dll':
```
Comment (jeremymeng) [inline vscode-extension/src/extension.ts:199]: I consider winrt and netmodule libraries too.
```suggestion
        case '.dll':
        case '.winrt':
        case '.netmodule':
```

File: vscode-extension/src/extension.ts [hunk 2]
```diff
@@ -173,3 +176,51 @@ async function promptForAssemblyFilePathViaDialog(): Promise<string> {
...
+          fileName: p.base,
+          fileExtension: p.ext,
+          fullPath: assemblyPath,
+          relativePath: p.dir.replace(assemblyWorkspace.uri.fsPath, ''),
+          workspaceFolder: assemblyWorkspace.name
+        };
+    });
+}
+
+function createAssemblyQuickPickItem(assemblyPathInfo: AssemblyPathInfo): vscode.QuickPickItem{
```
Comment (jeremymeng) [inline vscode-extension/src/extension.ts:196]:
```suggestion
function createAssemblyQuickPickItem(assemblyPathInfo: AssemblyPathInfo): AssemblyQuickPickItem{
```

File: vscode-extension/src/extension.ts [hunk 3]
```diff
@@ -38,8 +38,8 @@ export function activate(context: vscode.ExtensionContext) {
     // The commandId parameter must match the command field in package.json
     disposables.push(vscode.commands.registerCommand('ilspy.decompileAssemblyInWorkspace', async () => {
         // The code you place here will be executed every time your command is executed
-        const assembly = await pickAssembly();
-        await decompileFile(assembly);
+        const assembly = await pickAssembly() as AssemblyQuickPickItem;
```
Comment (jeremymeng) [inline vscode-extension/src/extension.ts:41]: You can return `AssemblyQuickPickItem` from `pickAssembly()` then casting is not needed.
```suggestion
        const assembly = await pickAssembly();
```

File: vscode-extension/src/extension.ts [hunk 4]
```diff
@@ -133,9 +133,12 @@ function showCodeInEditor(code: string, language: string, viewColumn: vscode.Vie
     writeStream.end();
 }
 
-async function pickAssembly(): Promise<string> {
+async function pickAssembly(): Promise<vscode.QuickPickItem> {
```
Comment (jeremymeng) [inline vscode-extension/src/extension.ts:136]:
```suggestion
async function pickAssembly(): Promise<AssemblyQuickPickItem> {
```

File: vscode-extension/src/extension.ts [hunk 5]
```diff
@@ -133,9 +133,12 @@ function showCodeInEditor(code: string, language: string, viewColumn: vscode.Vie
...
     const assemblies = await findAssemblies();
-    return await vscode.window.showQuickPick(assemblies);
+    const assemblyPathInfo: AssemblyPathInfo[] = parseAssemblyPath(assemblies);
+    const quickPickItems = assemblyPathInfo.map(
+      info => createAssemblyQuickPickItem(info));
+    return await vscode.window.showQuickPick<vscode.QuickPickItem>(quickPickItems);
```
Comment (jeremymeng) [inline vscode-extension/src/extension.ts:141]:
```suggestion
    return await vscode.window.showQuickPick<AssemblyQuickPickItem>(quickPickItems);
```

File: vscode-extension/src/extension.ts [hunk 6]
```diff
@@ -173,3 +176,51 @@ async function promptForAssemblyFilePathViaDialog(): Promise<string> {
...
+          fileExtension: p.ext,
+          fullPath: assemblyPath,
+          relativePath: p.dir.replace(assemblyWorkspace.uri.fsPath, ''),
+          workspaceFolder: assemblyWorkspace.name
+        };
+    });
+}
+
+function createAssemblyQuickPickItem(assemblyPathInfo: AssemblyPathInfo): vscode.QuickPickItem{
+    const selectIcon = (extension: string) => {;
```
Comment (jeremymeng) [inline vscode-extension/src/extension.ts:197]: extraneous `;` at end it looks.
```suggestion
    const selectIcon = (extension: string) => {
```

File: vscode-extension/src/extension.ts [hunk 7]
```diff
@@ -173,3 +176,51 @@ async function promptForAssemblyFilePathViaDialog(): Promise<string> {
...
+
+interface AssemblyPathInfo {
+    fullPath: string;
+    relativePath: string;
+    fileName: string;
+    fileExtension: string;
+    workspaceFolder?: string;
+}
+
+interface AssemblyQuickPickItem extends vscode.QuickPickItem{
```
Comment (jeremymeng) [inline vscode-extension/src/extension.ts:224]: nitpick: space
```suggestion
interface AssemblyQuickPickItem extends vscode.QuickPickItem {
```

**Comment by NikiforovAll** [comment]
Thanks for your review.

**Comment by jeremymeng** [comment]
Thank you for the contribution @NikiforovAll!

**Comment by NikiforovAll** [comment]
Thank you! 🙏

**Pull request merged**